### ISSUE TITLE
Multiple Cookie Header Fields in recorded ZEST script

### ISSUE THREAD

**Issue by derpsteb** [zaproxy/zaproxy#4757]
**Describe the bug**
When using a recorded ZEST script to authenticate against the DVWA, ZAP sets multiple cookie header fields instead of a single cookie list field (https://tools.ietf.org/html/rfc6265#section-5.4).

**To Reproduce**
1. Record ZEST script that include : 
GET to dvwa/login.php, 
POST to dvwa/login.php
2. Run script 
3. Response to 1st request should include 2 set-cookie headers
4. POST request now includes both cookies, but as separate ones like:
```
Cookie: PHPSESSID=l7ocif51gpla5hjp7bfqht0ki7,
Cookie: security=low
```

**Expected behavior**
The included cookie in the POST request header should be:
` Cookie: PHPSESSID=l7ocif51gpla5hjp7bfqht0ki7; security=low `

**Software versions**
 - ZAP: 2.7.0
 - Add-on: Zest 26.0.0
 - OS: Manjaro 17.1 with kernel 4.9
 - Java: 1.8.0_172
 - Browser: Firefox 60.0.2

**Additional context**
It seems like @thc202 already fixed the issue with [this](https://github.com/thc202/zaproxy/releases/tag/issue-cookies-zest-3) jar.

**Comment by thc202** [comment]
Thanks for raising the issue!

**Comment by shankbond** [comment]
Hi all,

Any expected fix date/workaround for this issue?

**Comment by kingthorin** [comment]
ZAP is an Open Source project maintained by a group of volunteers, at this time I'm not aware of anyone actively working on this issue. We're always happy to accept contributions from the community.

You could also try putting a bounty on the issue and seeing if it gets any attention: https://www.bountysource.com/issues/59625607-multiple-cookie-header-fields-in-recorded-zest-script

**Comment by thc202** [comment]
Actually this is already fixed in the zap.jar linked from the issue (for a workaround), I didn't open a PR yet because I still need to check whether this is better fixed in Zest library or in ZAP.

**Comment by jgauthier78** [comment]
> Actually this is already fixed in the zap.jar linked from the issue (for a workaround), I didn't open a PR yet because I still need to check whether this is better fixed in Zest library or in ZAP.

Hello, I was attending a training based on ZAP 2.7.0 talking about this issue.
I'm not sure to understand what was the output on Dec 2018.
@thc202  did you eventually open a PR?
@derpsteb I guess that if you reopened this issue it means that it still exists in 2.9.0 version correct?
As commented by @kingthorin I could potentially help but am not sure about the best way to proceed since @thc202 already fixed the issue on 2018 :)

**Comment by derpsteb** [comment]
I am very sorry for reopening the issue, that was just a misclick. I don't know the current status. I just thought that the problem may still exist and you might want to keep track of it. But feel free to close the issue :)

**Comment by thc202** [comment]
This was fixed in the Zest library (with mozilla/zest#168) and the Zest add-on is already using the newer Zest library (#4797).
